Lib/Protocols/IdSSLOpenSSL.pas

```diff
@@ -3567,23 +3575,30 @@ procedure TIdSSLContext.InitContext(CtxMode: TIdSSLCtxMode);
   //set SSL Versions we will use
   if HasTLS_method then
   begin
-    if sslvSSLv3 in SSLVersions then
-      SSL_CTX_set_min_proto_version(fContext,SSL3_VERSION)
-    else
-    if sslvTLSv1 in SSLVersions then
-      SSL_CTX_set_min_proto_version(fContext,TLS1_VERSION)
-    else
-    if sslvTLSv1_1 in SSLVersions then
-      SSL_CTX_set_min_proto_version(fContext,TLS1_1_VERSION)
-    else
-    if sslvTLSv1_2 in SSLVersions then
-      SSL_CTX_set_min_proto_version(fContext,TLS1_2_VERSION)
-    else
-    if sslvTLSv1_3 in SSLVersions then
-      SSL_CTX_set_min_proto_version(fContext,TLS1_3_VERSION)
-    else
-    {Set to the lowest now possible}
-      SSL_CTX_set_min_proto_version(fContext,SSL3_VERSION);
+    if SSLVersions <> [] then
+    begin
+      for v := sslvSSLv3 to high(TIdSSLVersions) do

```

Comment:
In Delphi (XE7, 11) compilation fails with "E2008 Incompatible types".
High/Low functions does not work with set types.